### ISSUE TITLE
Group-Role Assignment not working in GDAP Bulk Tool but works in Partner Center

### ISSUE THREAD

**Issue by sansbacher** [microsoft/PartnerCenter-GDAPTransition#19]
Hi,

I have setup the GDAP Bulk Migration Tool - using just 2 of our Customers as a test, and 3 new Groups I created in our AAD. I assigned 2 AAD Roles to two of the Groups and 20+ Roles to the final Group. I ran the Tool, it worked (created Relationships and assigned Roles to Groups).

**EXCEPT**: one of the Groups (the one with multiple Roles, called "Service Admins") is sitting at Status = "Pending" in `accessAssignment.csv`, I left it over the weekend and still it is Pending. I check in the web GUI (MS Partner Center) and it's Pending. No errors or warnings appear when using the Tool.

I Terminated the Relationships, trimmed the number of Roles for that 3rd Group down to just 16 and tried again: exactly the same, the "Service Admins" Group is Pending.

So I tried making a new AAD Group, and _manually_ assigned the _same_ 16 Roles to the Group via Partner Center, a few at a time: it worked! I Deleted the Group and re-added the Group and all 16 Roles at once and it worked!

Is there some issue with **assigning multiple Roles to Groups in the GDAP Migration Tool but not via the website**? The list of Roles shows up for the Pending Group, but the Status remains Pending.

**Has anyone been able to assign multiple (10+) Roles to a single Group with the Tool?**

If it matters, the 16 roles I tried to assign were: Global reader, Compliance administrator, Compliance data administrator, Conditional access administrator, Exchange administrator, Exchange recipient administrator, Groups administrator, Helpdesk administrator, License administrator, Office apps administrator, Password administrator, Power platform administrator, Search administrator, Service support administrator, SharePoint administrator, User administrator.

I found the Gbm\Logs folder, but nothing useful in there about this. I am using CSV not JSON.

--Saul

**Comment by manusidd** [comment]
Hello Saul,
I verified the access assignment for same set of roles that you have mentioned, and the assignments were created and activated successfully for all the roles. I used the CSV mode and with roles names exactly how its mentioned above.
I would like you to confirm the following:
1. All the roles were part of ADRoles.csv under the operations folder
2. Relationships were also created with all these roles already configured in the ADRoles.csv that is under operations folder.
Attached how the securityGroup.csv file looks for me.

![image](https://user-images.githubusercontent.com/100112615/189611343-b0d80dfb-8089-4059-a79c-b22bd7c98b47.png)

Let me know.
Manu

**Comment by sansbacher** [comment]
Thanks Manu,

I have managed to solve my problem and assignment of GDAP Roles to Groups is now working! 

All of the Roles I was trying to assign were included in `operations\ADRoles.csv`.
However I was using Role IDs not Role Names, as directed in the GDAP Bulk Migration Tool [instructions](https://docs.microsoft.com/en-us/partner-center/gdap-bulk-migration-tool#provision-security-groups) ("CommaSeperatedRoles – A list that must contain one or more (separated by commas) of the **Azure AD role IDs** that were defined in the ADRoles.csv file that were assigned to the newly created GDAP relationships.")

I tried using the Role Names as you had and experienced the same result (adding 2 Roles worked, but trying to add the bunch left it Pending).

Then I cross-referenced all Roles I was trying to add (and all the Roles in `ADRoles.csv`) with the Roles listed in `Configuration\ADRolesTemplatesAll.json` and I found I had an _extra_ "Role" called "User"! I removed that entry from `ADRoles.csv` and from the **CommaSeperatedRoles** column in `operations\securityGroup.csv`, Terminated the existing GDAP relationships, and started from scratch.

All show up as Status = "active" now!

I believe I found the "User" role because initially I hadn't found `ADRoleTemplatesAll.json` as the master list of all possible Roles, so I simply used the results from `Get-AzureADDirectoryRoleTemplate` which lists "User" as "Default role for member users. Can read all and write a limited set of directory information." But I guess that doesn't need to be added :)

**For anyone else finding this:** only use Roles from `ADRoleTemplatesAll.json` or from [Azure AD built-in roles](https://docs.microsoft.com/en-us/azure/active-directory/roles/permissions-reference#all-roles).

  --Saul
